src/polygon2d.jl

```diff
@@ -0,0 +1,30 @@
+"""
+```
+draw_polygon_perimeter!(img, vertices)
+draw_polygon_perimeter!(img, vertices, color)
+img2=draw_polygon_perimeter(img, vertices)
+img2=draw_polygon_perimeter(img, vertices, color)
+```
+Draws the perimeter of the polygon formed by input vertices(Array of CartesianIndex).
+"""
+
+draw_polygon_perimeter{T<:Colorant}(img::AbstractArray{T, 2}, args...) = draw_polygon_perimeter!(copy(img), args...)
+
+draw_polygon_perimeter!{T<:Colorant}(img::AbstractArray{T, 2}, vertices::Array{CartesianIndex{2},1})=draw_polygon_perimeter!(img::AbstractArray{T, 2}, vertices::Array{CartesianIndex{2},1}, one(T))
+
+function draw_polygon_perimeter!{T<:Colorant}(img::AbstractArray{T, 2}, vertices::Array{CartesianIndex{2},1}, color::T)
+
+    f = CartesianIndex(map(r->first(r)-1, indices(img)))
+    l = CartesianIndex(map(r->last(r), indices(img)))
+    for vertex in vertices
+        if min(f, vertex)!=f || max(l, vertex)!=l
+            println(vertex)
+            error("Polygon coordinates out of range.")
+        end
+    end
```

Comment:
I don't like warnings; suppose the user deliberately wanted your second option, it would be really annoying to have a warning pop up each time. I think if folks want that version, we should support a "clipping path" that breaks the path up into in-bounds segments that terminate when they get to the edge:
```julia
segments = clip(vertices, indices(img))
for seg in segments
    polygon_perimiter!(img, seg)
end
```
Otherwise, I think we should simply throw an error if there's an out-of-bounds segment. We could either do that
- before drawing anything at all
- as soon as the first out-of-bounds vertex is encountered (which terminates future edges even if they are in bounds)
- your option 1

Of these, I like the first and third better than the middle one. The middle one would be the absolute highest performance option, but compared to the cost of actually drawing the pixels I doubt we would notice.